applications/HDF5Application/custom_python/add_custom_io_to_python.cpp

```diff
@@ -29,45 +28,192 @@
 // Application includes
 #include "custom_io/hdf5_file.h"
 #include "custom_io/hdf5_model_part_io.h"
-#include "custom_io/hdf5_nodal_solution_step_data_io.h"
-#include "custom_io/hdf5_element_data_value_io.h"
-#include "custom_io/hdf5_element_flag_value_io.h"
-#include "custom_io/hdf5_condition_data_value_io.h"
-#include "custom_io/hdf5_condition_flag_value_io.h"
-#include "custom_io/hdf5_nodal_solution_step_bossak_io.h"
-#include "custom_io/hdf5_nodal_data_value_io.h"
-#include "custom_io/hdf5_nodal_flag_value_io.h"
 #include "custom_io/hdf5_data_value_container_io.h"
-#include "custom_io/hdf5_element_gauss_point_output.h"
-#include "custom_io/hdf5_condition_gauss_point_output.h"
 #include "custom_io/hdf5_vertex_container_io.h"
+#include "custom_io/hdf5_container_component_io.h"
+
+#include "custom_utilities/container_io_utils.h"
+
 #ifdef KRATOS_USING_MPI
 #include "custom_io/hdf5_partitioned_model_part_io.h"
 #endif
 
 namespace Kratos {
 namespace Python {
 
+template<class TContainerType, class TContainerDataIOType>
+struct VariableContainerComponentIOWrapper
+{
+    using ContainerType = TContainerType;
+
+    using ContainerDataIOType = TContainerDataIOType;
+
+    using ContainerIOType = typename HDF5::ContainerComponentIO<
+                                        TContainerType,
+                                        TContainerDataIOType,
+                                        Variable<int>,
+                                        Variable<double>,
+                                        Variable<array_1d<double, 3>>,
+                                        Variable<array_1d<double, 4>>,
+                                        Variable<array_1d<double, 6>>,
+                                        Variable<array_1d<double, 9>>,
+                                        Variable<Kratos::Vector>,
+                                        Variable<Kratos::Matrix>>;
+
+};
+
+template<class TContainerType>
+struct FlagContainerComponentIOWrapper
+{
+    using ContainerType = TContainerType;
+
+    using ContainerDataIOType = HDF5::Internals::FlagIO;
+
+    using ContainerIOType = typename HDF5::ContainerComponentIO<
+                                        TContainerType,
+                                        ContainerDataIOType,
+                                        Flags>;
+
+};
+
+template<class TContainerComponentIOWrapper>
+void AddContainerComponentIOToPython(
+    pybind11::module& m,
+    const std::string& rName)
+{
+    namespace py = pybind11;
+
+    using data_io = typename TContainerComponentIOWrapper::ContainerIOType;
+
+    using container_type = typename TContainerComponentIOWrapper::ContainerType;
 
+    using container_data_io_type = typename TContainerComponentIOWrapper::ContainerDataIOType;
+
+    std::string container_name, arg_name;
+    if constexpr(std::is_same_v<container_type, ModelPart::NodesContainerType>) {
+        container_name = "Nodal";
```

Comment:
Nodal is inconsistent with Element and Condition

Reply:
Yep, but I need it to have backward compatibility with the legacy HDF5 writing mechanism. :/